packages/sdk/src/dataProtector/sharing/removeProtectedDataAsRentable.ts

```diff
@@ -0,0 +1,41 @@
+import { ethers, type Contract } from 'ethers';
+import { ABI as sharingABI } from '../../contracts/sharingAbi.js';
+import { WorkflowError } from '../../utils/errors.js';
+import { throwIfMissing } from '../../utils/validators.js';
+import {
+  AddressOrENS,
+  IExecConsumer,
+  RemoveProtectedDataAsRentableParams,
+  RemoveProtectedDataAsRentableResponse,
+} from '../types.js';
+
+export const removeProtectedDataAsRentable = async ({
+  iexec = throwIfMissing(),
+  collectionTokenId = throwIfMissing(),
+  protectedDataAddress = throwIfMissing(),
+  sharingContractAddress,
+}: IExecConsumer & {
+  sharingContractAddress: AddressOrENS;
+} & RemoveProtectedDataAsRentableParams): Promise<RemoveProtectedDataAsRentableResponse> => {
+  //TODO:Input validation
+  const { provider, signer } = await iexec.config.resolveContractsClient();
+
```

Comment:
In order to be more confident about the following smart-contract call, we could check a few things here:

 - checkProtectedData() →
. Check that the protected data exists
. Check that the protected data is owned by the Sharing smart-contract
. Check that the protected data is in a collection owned by the user

 - checkCollection() →
. Check that the collection exists
. Check that the collection is owned by the user

and throw an explicit error if necessary, which would make it clearer for the builder.
I've done something similar for `addToCollection()`, throwing an `ErrorWithData` for example.
Tell me what you think.

Reply:
Yeah i agree. In fact, you should implement all the require + modifier that is implemented in this function on the SC to avoid TX that will revert and return the appropriate message

Reply:
i added 2 checks : isCollectionOwner & isProtectedDataInCollection